src/client/InteractionClient.js

```diff
@@ -0,0 +1,178 @@
+'use strict';
+
+const BaseClient = require('./BaseClient');
+const APIMessage = require('../structures/APIMessage');
+const Interaction = require('../structures/Interaction');
+const { ApplicationCommandOptionType, InteractionType, InteractionResponseType } = require('../util/Constants');
+
+let sodium;
+
+/**
+ * Interaction client is used for interactions.
+ *
+ * ```js
+ * const client = new InteractionClient({
+ *   token: ABC,
+ *   publicKey: XYZ,
+ * }, async (interaction) => {
+ *   if (will take a long time) {
+ *     doSomethingLong.then((d) => {
+ *       interaction.reply({
+ *         content: 'wow that took long',
+ *       });
+ *     });
+ *     // return null to signal that we will be replying via `interaction.reply`.
+ *     return null;
+ *   }
+ *   return { content: 'hi!' };
+ * });
+ * ```
+ */
+class InteractionClient extends BaseClient {
+  /**
+   * @param {Options} options Options for the client.
+   * @param {Handler} handler Handler to handle things.
+   * @param {undefined} client For internal use.
```

Comment:
Why is it `undefined` ? Also why isn't it the first args for consistency ?

Reply:
`Discord.InteractionClient` is a public class which can be used without the main `Discord.Client` class.